docker-compose.yml

```diff
@@ -0,0 +1,65 @@
+version: "3"
+
+services:
+  ngrok:
+    container_name: ngrok
+    expose:
+      - "4040"
+    ports:
+      - "4040:4040"
+    image: wernight/ngrok
+    environment:
+      - NGROK_PORT=server:3000
+
+  postgres:
+    image: postgres:10-alpine
+    environment:
+      - POSTGRES_USER=postgres
+      - POSTGRES_PASSWORD=root
+    volumes:
+      - postgresqldata:/var/lib/postgresql/data
+    expose:
+      - "5432"
+    healthcheck:
+      test: ["CMD-SHELL", "pg_isready -U postgres"]
+      interval: 5s
+      timeout: 5s
+      retries: 5
+
+  redis:
+    image: redis:5-alpine
+    expose:
+      - "6379"
+    volumes:
+      - redisdata:/data
+
+  server:
+    build: .
+    volumes:
+      - .:/myapp
+    command: >
+      sh -c "sleep 10 &&
```

Comment:
The `sleep` is for?

Reply:
I realised that the at least one of the services that the server depends on (I think postgres) takes a few seconds to start fully once the containers spin up, but the 'depends on' docker compose feature is satisfied as the service is technically running. The sleep basically pauses the server trying to connect until all the other service should be ready - but I admit it is a bit of a hack :)

Reply:
No problem, these kind of things are acceptable in setup scripts I feel :)